### ISSUE TITLE
NRE in ProjectSystem.VS.References.DesignTimeAssemblyResolution.GetAllResolvedReferences

### ISSUE THREAD

**Issue by tmat** [dotnet/project-system#1973]
When adding a P2P reference.

```
=====================
4/11/2017 1:53:48 PM
Crippling
System.AggregateException: One or more errors occurred. ---> System.ArgumentNullException: Value cannot be null.
Parameter name: key
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at Microsoft.VisualStudio.ProjectSystem.VS.References.DesignTimeAssemblyResolution.GetAllResolvedReferences()
   at Microsoft.VisualStudio.ProjectSystem.VS.References.DesignTimeAssemblyResolution.ResolveReferences(String[] originalNames, AssemblyName[] assemblyName, VsResolvedAssemblyPath[] assemblyPaths)
   at Microsoft.VisualStudio.ProjectSystem.VS.References.DesignTimeAssemblyResolution.ResolveAssemblyPathInTargetFx(String[] prgAssemblySpecs, UInt32 cAssembliesToResolve, VsResolvedAssemblyPath[] prgResolvedAssemblyPaths, UInt32& pcResolvedAssemblyPaths)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.ResolveAssemblyPath(String assemblySpec)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.ResolveAssembly(AssemblyName assemblyName)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.GetSystemAssembly()
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse..ctor(IDesignTimeAssemblyLoader assemblyLoader, IVsDesignTimeAssemblyResolution projectAssemblyResolution, IVsSmartOpenScope dispenser)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.GetUniverse(IDesignTimeAssemblyLoader assemblyLoader, IVsDesignTimeAssemblyResolution projectAssemblyResolution, IVsSmartOpenScope dispenser)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkProvider..ctor(IVsDesignTimeAssemblyResolution assemblyResolution, IDesignTimeAssemblyLoader assemblyLoader, TypeDescriptionProvider parentProvider, IVsSmartOpenScope openScope)
   at Microsoft.VisualStudio.ProjectUtilities.GetTargetFrameworkProvider(IServiceProvider provider, IDesignTimeAssemblyLoader assemblyLoader, IVsHierarchy hierarchy, Object instance)
   at Microsoft.VisualStudio.Design.VSTypeResolutionService.get_TypeProvider()
   at Microsoft.VisualStudio.Design.VSTypeResolutionService.VSLangProj._dispReferencesEvents.ReferenceAdded(Reference r)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.Automation.OAVSReferences.OnReferenceAdded(Reference reference)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ReferencesHostBridge.ApplyAsync(ReferencesDelta value)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectHostBridge`3.<>c__DisplayClass36_1.<<InitializeCoreAsync>b__3>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Threading.JoinableTask.<JoinAsync>d__76.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Threading.JoinableTask`1.<JoinAsync>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectHostBridge`3.<>c__DisplayClass36_0.<<InitializeCoreAsync>b__2>d.MoveNext()
   --- End of inner exception stack trace ---
---> (Inner Exception #0) System.ArgumentNullException: Value cannot be null.
Parameter name: key
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at Microsoft.VisualStudio.ProjectSystem.VS.References.DesignTimeAssemblyResolution.GetAllResolvedReferences()
   at Microsoft.VisualStudio.ProjectSystem.VS.References.DesignTimeAssemblyResolution.ResolveReferences(String[] originalNames, AssemblyName[] assemblyName, VsResolvedAssemblyPath[] assemblyPaths)
   at Microsoft.VisualStudio.ProjectSystem.VS.References.DesignTimeAssemblyResolution.ResolveAssemblyPathInTargetFx(String[] prgAssemblySpecs, UInt32 cAssembliesToResolve, VsResolvedAssemblyPath[] prgResolvedAssemblyPaths, UInt32& pcResolvedAssemblyPaths)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.ResolveAssemblyPath(String assemblySpec)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.ResolveAssembly(AssemblyName assemblyName)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.GetSystemAssembly()
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse..ctor(IDesignTimeAssemblyLoader assemblyLoader, IVsDesignTimeAssemblyResolution projectAssemblyResolution, IVsSmartOpenScope dispenser)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkUniverse.GetUniverse(IDesignTimeAssemblyLoader assemblyLoader, IVsDesignTimeAssemblyResolution projectAssemblyResolution, IVsSmartOpenScope dispenser)
   at Microsoft.VisualStudio.Shell.Design.VsTargetFrameworkProvider..ctor(IVsDesignTimeAssemblyResolution assemblyResolution, IDesignTimeAssemblyLoader assemblyLoader, TypeDescriptionProvider parentProvider, IVsSmartOpenScope openScope)
   at Microsoft.VisualStudio.ProjectUtilities.GetTargetFrameworkProvider(IServiceProvider provider, IDesignTimeAssemblyLoader assemblyLoader, IVsHierarchy hierarchy, Object instance)
   at Microsoft.VisualStudio.Design.VSTypeResolutionService.get_TypeProvider()
   at Microsoft.VisualStudio.Design.VSTypeResolutionService.VSLangProj._dispReferencesEvents.ReferenceAdded(Reference r)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.Automation.OAVSReferences.OnReferenceAdded(Reference reference)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ReferencesHostBridge.ApplyAsync(ReferencesDelta value)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectHostBridge`3.<>c__DisplayClass36_1.<<InitializeCoreAsync>b__3>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Threading.JoinableTask.<JoinAsync>d__76.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.Threading.JoinableTask`1.<JoinAsync>d__3.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at Microsoft.VisualStudio.ProjectSystem.VS.Implementation.Package.ProjectHostBridge`3.<>c__DisplayClass36_0.<<InitializeCoreAsync>b__2>d.MoveNext()<---

===================

```

**Comment by davkean** [comment]
Thanks - this is known and fixed in 15.1. What build are you running?

**Comment by tmat** [comment]
The VS internal channel dogfood build.
